modules/formulize/include/formdisplay.php

```diff
@@ -241,85 +234,35 @@ public function renderValidationJS($withtags = true) {
-
-        // cache also the screen id since we need to lookup the template according to the screen object settings
-
-        // if a numeric value is passed as ele, that is an element id, sent specifically so we can seed the right column value based on the screen setting
-        if(is_numeric($ele)) {
-            $eleKey = $ele;
-        } elseif(is_object($ele) AND isset($ele->formulize_element)) {
-            $eleKey = $ele->formulize_element->getVar('ele_id');
-        } elseif(is_object($ele)) {
-			$eleKey = md5(serialize($ele));
-        } elseif(strstr($ele, "<<||>>")) {
-            $ele = explode("<<||>>", $ele);
-            $parts = explode('_', $ele[1]);
-            $ele_id = $parts[3];
-            $eleKey = $ele_id;
-        } else {
-            $eleKey = md5($ele);
-        }
-
-        if(isset($_SESSION['columns'][$uid][$eleKey]) AND !$reset) {
-            $columns = $_SESSION['columns'][$uid][$eleKey];
-        } elseif($this->getTitle() != 'formulizeAsynchElementRender' AND is_object($this->screen)) {
-            $_SESSION['formulizeScreenId'][$uid] = $this->screen->getVar('sid');
-            $columns = $this->screen->getVar('displaycolumns') == 1 ? 1 : 2;
-            if($this->screen->getVar('column1width')) {
-                $column1width = $this->screen->getVar('column1width');
-            } elseif($columns == 2) {
-                $column1width = '20%';
-            } else {
-                $column1width = 'auto';
-            }
-            $column2width = $this->screen->getVar('column2width') ? $this->screen->getVar('column2width') : 'auto';
-            $columns = array($columns, $column1width, $column2width);
-        } elseif($_SERVER['SCRIPT_NAME'] == '/modules/formulize/admin/fakeform.php') {
-						$columns = array(2, '20%', 'auto');
-        } else {
-            $columns = array(1, 'auto', 'auto');
-        }
-        $_SESSION['columns'][$uid][$eleKey] = $columns;
-        return $columns;
+    function _getColumns() {
+			if(is_object($this->screen)) {
+					$columns = $this->screen->getVar('displaycolumns') == 1 ? 1 : 2;
+					if($this->screen->getVar('column1width')) {
+							$column1width = $this->screen->getVar('column1width');
+					} elseif($columns == 2) {
+							$column1width = '20%';
+					} else {
+							$column1width = 'auto';
+					}
+					$column2width = $this->screen->getVar('column2width') ? $this->screen->getVar('column2width') : 'auto';
+					$columns = array($columns, $column1width, $column2width);
+			} elseif($_SERVER['SCRIPT_NAME'] == '/modules/formulize/admin/fakeform.php') {
+					$columns = array(2, '20%', 'auto');
+			} else {
+					$columns = array(1, 'auto', 'auto');
+			}
+			return $columns;
     }
 
 	function _drawElements($elements, $ret, $hidden) {
 
 		foreach ( $elements as $ele ) {
 
-            if(is_string($ele) AND isset($GLOBALS['formulize_renderedElementHasConditions'])) {
```

Comment:
Awkward stuff that was being used to call _getColumns to basically cache the data in the session for all elements even if not rendered this page load, so we were ready for conditional rendering. Yucky. All gone now!